spec/index.md

```diff
@@ -6282,7 +6282,7 @@ E.content = CanisterQuery Q
 Q.canister_id ∈ verify_envelope(E, Q.sender, S.system_time)
 |Q.nonce| <= 32
 is_effective_canister_id(E.content, ECID)
-S.system_time <= Q.ingress_expiry
+S.system_time <= Q.ingress_expiry or Q.sender = anonymous_id
```

Comment:
Obviously this is a complication if we have yet another condition. 
I seriously don't know why we need to change the spec instead of relaxing the checks inside the replica without changing the spec.